src/csrc/include/internal/model_state.h

```diff
@@ -40,6 +40,7 @@ namespace torchfort {
 struct ModelState {
   int64_t step_train;
   int64_t step_inference;
+  int64_t step_train_current; // training step of current run (ignoring restarted state)
```

Comment:
why do we need this additional counter btw? Can step_train not be used in place of step_train_current?

Reply:
This is handle situations where one might save a checkpoint at a training step that is not a multiple of `grad_accumulation_steps` (e.g. say they save a checkpoint at `step_train = 10` with `grad_accumulation_steps = 4`). Now, if they were to restart, if we used `step_train` for our gradient accumulation timing, we would perform a weight update after 2 steps, which is not correct. `step_train_current` always begins at zero regardless of the restarted state, so we can use that to ensure we perform weight updates at the correct interval.

Reply:
That makes total sense, thank you.